platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/util/LabelProvider.kt

```diff
@@ -0,0 +1,15 @@
+// Copyright 2000-2025 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.jewel.ui.util
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.staticCompositionLocalOf
+import org.jetbrains.jewel.foundation.InternalJewelApi
+
+@InternalJewelApi
+public interface LabelProvider {
```

Comment:
I'm not sure about the `LabelProvider` name. Label usually represents the UI element on screen to me, not the text.

How about `TextProvider` or `StringProvider`? `StringProvider` would be a good match in the future if we add compose resources and localize the messages properly.